spec.bs

```diff
@@ -1753,20 +1751,21 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
               [=trusted bidding signals batcher/data versions=][|igName|].
             1. Set |allTrustedBiddingSignals| to [=trusted bidding signals batcher/all trusted bidding signals=].
           1. Otherwise if |trustedBiddingSignalsBatcher|'s
-            [=trusted bidding signals batcher/no signals flags=][|igName|] is "fetch-failed", then
-            [=add a platform contribution=] with [=trusted bidding signals failure bucket=],
-            |realTimeContributionsMap|, and |igName|.
+            [=trusted bidding signals batcher/no signals flags=][|igName|] is "fetch-failed", and
+            |optedInForRealTimeReporting| is true, then:
+            1. [=Add a platform contribution=] with [=trusted bidding signals failure bucket=],
+              |realTimeContributionsMap|, and |igName|.
           1. If |dataVersion| is not null, then [=map/set=] |browserSignals|["{{BiddingBrowserSignals/dataVersion}}"] to |dataVersion|.
           1. Otherwise, [=map/remove=] |browserSignals|["{{BiddingBrowserSignals/dataVersion}}"].
           1. Let |generateBidResult| be the result of running [=generate potentially multiple bids=]
             with |allTrustedBiddingSignals|, |auctionSignals|, a [=map/clone=] of |browserSignals|,
             |perBuyerSignals|, |directFromSellerSignalsForBuyer|, |perBuyerTimeout|,
             |expectedCurrency|, |multiBidLimit|, |ig|, and |auctionStartTime|.
           1. If |generateBidResult| is failure, then:
-            1. [=add a platform contribution=] with [=bidding script failure bucket=],
-              |realTimeContributionsMap| and |buyer|.
+            1. If |optedInForRealTimeReporting| is true, then [=add a platform contribution=] with

```

Comment:
So looking at how it works, I noticed the spec doesn't hard fail when WASM loading fails, which the implementation does, I think.